step-plans/step-plans-base-artefacts/src/main/java/step/artefacts/handlers/functions/TokenAutoscalingExecutionPlugin.java

```diff
@@ -53,39 +54,70 @@ public void initializeExecutionEngineContext(AbstractExecutionEngineContext pare
 
     @Override
     public void provisionRequiredResources(ExecutionContext context) {
-        super.provisionRequiredResources(context);
-
-        // Get the results of the token forecasting
-        TokenForecastingContext tokenForecastingContext = getTokenForecastingContext(context);
-        Map<String, Integer> tokenForecastPerPool = tokenForecastingContext.getTokenForecastPerPool();
-        logger.debug("Token forecast estimation result: " + tokenForecastPerPool);
-
-        Set<Map<String, Interest>> criteriaWithoutMatch = tokenForecastingContext.getCriteriaWithoutMatch();
-        if (!criteriaWithoutMatch.isEmpty()) {
-            String message = "No matching pool found for selection criteria: " + criteriaWithoutMatch;
-            logger.warn(message);
-            throw new PluginCriticalException(message);
+        if (isAutoscalingEnabled()) {
```

Comment:
@david-stephan The 2 plugins could be split, forcasting should always be active and OS side, while AutoScaling could be  a licensed plugin on EE side.